src/libraries/System.Private.DataContractSerialization/src/System/Runtime/Serialization/Json/JsonEncodingStreamWrapper.cs

```diff
@@ -174,15 +174,18 @@ public override void Flush()
             _stream.Flush();
         }
 
-        public override int Read(byte[] buffer, int offset, int count)
+        public override int Read(byte[] buffer, int offset, int count) =>
+            Read(new Span<byte>(buffer, offset, count));
+
+        public override int Read(Span<byte> buffer)
```

Comment:
Can/should we override Write(span) on each of these streams as well?

Are there other streams in dotnet/runtime for which we should provide these overrides?

Reply:
> Can/should we override Write(span) on each of these streams as well?

Maybe? It could prevent future regressions, like the one I made here.

Looking at where these streams are used for writing, they only get instantiated when the Encoding is not UTF8 (which is different than read - where they always are used). And looking through all the "Write" methods to these streams, they all pass an array, offset, and count. I don't think I should add it since they will never get used. Thoughts?

> Are there other streams in dotnet/runtime for which we should provide these overrides?

I looked through all the places that override `Read(byte[] ...)` and looked to see if they override `Read(Span)` as well. 

### Here's are ones I think we should add the additional override to:

- **Passed to `BinaryReader` / XmlReader / other place I changed to call ReadAtLeast / ReadExactly:**

https://github.com/dotnet/runtime/blob/472050d35cbbfb90c2d0372c97bbc3d271c5e68a/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/Design/DesigntimeLicenseContextSerializer.cs#L87
https://github.com/dotnet/runtime/blob/472050d35cbbfb90c2d0372c97bbc3d271c5e68a/src/libraries/System.Data.Common/src/System/Data/SQLTypes/SqlXml.cs#L406
https://github.com/dotnet/runtime/blob/472050d35cbbfb90c2d0372c97bbc3d271c5e68a/src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/ImmutableMemoryStream.cs#L67
https://github.com/dotnet/runtime/blob/472050d35cbbfb90c2d0372c97bbc3d271c5e68a/src/libraries/System.Reflection.Metadata/src/System/Reflection/Internal/Utilities/ReadOnlyUnmanagedMemoryStream.cs#L31

- **These get returned through public methods, so they probably should:**

https://github.com/dotnet/runtime/blob/472050d35cbbfb90c2d0372c97bbc3d271c5e68a/src/libraries/System.Data.Common/src/System/Data/SQLTypes/SQLBytes.cs#L695
https://github.com/dotnet/runtime/blob/472050d35cbbfb90c2d0372c97bbc3d271c5e68a/src/libraries/System.IO.Packaging/src/System/IO/Packaging/ZipWrappingStream.cs#L61
https://github.com/dotnet/runtime/blob/472050d35cbbfb90c2d0372c97bbc3d271c5e68a/src/libraries/System.Net.Requests/src/System/Net/FtpDataStream.cs#L175

### Here are the ones I'm not sure we need to:

- **Only appears to be used by `FtpControlStream`, I can't tell if anyone ever calls Read with a Span. Looks easy to do though.**

https://github.com/dotnet/runtime/blob/472050d35cbbfb90c2d0372c97bbc3d271c5e68a/src/libraries/System.Net.Requests/src/System/Net/NetworkStreamWrapper.cs#L137

- **Has a lot of derived Stream classes, but I don't see them ever getting `Read(Span)` call on them. This also looks easy to do.**

https://github.com/dotnet/runtime/blob/472050d35cbbfb90c2d0372c97bbc3d271c5e68a/src/libraries/System.Net.Mail/src/System/Net/DelegatedStream.cs#L131

### Here's the ones I don't think we need to:

- **Only gets BeginRead and EndRead called on it**

https://github.com/dotnet/runtime/blob/472050d35cbbfb90c2d0372c97bbc3d271c5e68a/src/libraries/System.Net.HttpListener/src/System/Net/HttpRequestStream.cs#L16

- **Only used for writing**

https://github.com/dotnet/runtime/blob/472050d35cbbfb90c2d0372c97bbc3d271c5e68a/src/libraries/System.IO.Packaging/src/System/IO/Packaging/IgnoreFlushAndCloseStream.cs#L124

- **Only do "async" reading, so we can't do anything with a Span**

https://github.com/dotnet/runtime/blob/472050d35cbbfb90c2d0372c97bbc3d271c5e68a/src/libraries/System.Net.Security/src/System/Net/Security/NegotiateStream.cs#L305
https://github.com/dotnet/runtime/blob/472050d35cbbfb90c2d0372c97bbc3d271c5e68a/src/libraries/System.Net.Security/src/System/Net/Security/SslStream.cs#L741
https://github.com/dotnet/runtime/blob/472050d35cbbfb90c2d0372c97bbc3d271c5e68a/src/libraries/System.Security.Cryptography/src/System/Security/Cryptography/CryptoStream.cs#L292
https://github.com/dotnet/runtime/blob/472050d35cbbfb90c2d0372c97bbc3d271c5e68a/src/libraries/System.IO.Ports/src/System/IO/Ports/SerialStream.Unix.cs#L403

Reply:
I've added the overrides for the first 4 locations above, and 2 of the next 3. The SqlBytes one is a little more involved, so I decided to open a separate issue for it - https://github.com/dotnet/runtime/issues/69921.